src/backgroundblurprocessor/BackgroundBlurVideoFrameProcessor.ts

```diff
@@ -0,0 +1,257 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { loadWorker } from '../../libs/voicefocus/loader';
+import { supportsWASM, supportsWorker } from '../../libs/voicefocus/support';
+import { AssetSpec } from '../../libs/voicefocus/voicefocus';
+import DefaultBrowserBehavior from '../browserbehavior/DefaultBrowserBehavior';
+import ConsoleLogger from '../logger/ConsoleLogger';
+import Logger from '../logger/Logger';
+import LogLevel from '../logger/LogLevel';
+import ModelSpec from '../modelspec/ModelSpec';
+import Versioning from '../versioning/Versioning';
+import NoOpVideoFrameProcessor from '../videoframeprocessor/NoOpVideoFrameProcessor';
+import BackgroundBlurOptions from './BackgroundBlurOptions';
+import BackgroundBlurProcessor from './BackgroundBlurProcessor';
+import BackgroundBlurProcessorBuiltIn from './BackgroundBlurProcessorBuiltIn';
+import BackgroundBlurProcessorProvided from './BackgroundBlurProcessorProvided';
+import BlurStrength from './BackgroundBlurStrength';
+import BackgroundFilterPaths from './BackgroundFilterPaths';
+import BackgroundFilterSpec from './BackgroundFilterSpec';
+import ModelSpecBuilder from './ModelSpecBuilder';
+
+/** @internal */
+const CREATE_DEFAULT_MODEL_SPEC = (): ModelSpec =>
+  ModelSpecBuilder.builder().withSelfieSegmentationDefaults().build();
+
+/** @internal */
+const DEFAULT_CDN = 'https://static.sdkassets.chime.aws';
+
+/** @internal */
+const DEFAULT_PATHS: BackgroundFilterPaths = {
+  worker: `${DEFAULT_CDN}/bgblur/workers/worker.js`,
+  wasm: `${DEFAULT_CDN}/bgblur/wasm/_cwt-wasm.wasm`,
+  simd: `${DEFAULT_CDN}/bgblur/wasm/_cwt-wasm-simd.wasm`,
+};
+
+/**
+ * No-op implementation of the blur processor. An instance of this class will be returned when a user attempts
+ * to create a blur processor when it is not supported.
+ */
+/** @internal */
+class NoOpBackgroundBlurProcessor
+  extends NoOpVideoFrameProcessor
+  implements BackgroundBlurProcessor {
+  /**
+   * no-op
+   */
+  setBlurStrength(): void {}
+
+  /**
+   * no-op
+   * @returns
+   */
+  async loadAssets(): Promise<void> {
+    return;
+  }
+
+  /**
+   * no-op
+   */
+  addObserver(): void {}
+
+  /**
+   * no-op
+   */
+  removeObserver(): void {}
+}
+
+/**
+ * [[BackgroundBlurProcessorFactory]]
```

Comment:
[[BackgroundBlurProcessorFactory]] I guess would not link to anything as I could not find any file with that name, do you mean [[BackgroundBlurVideoFrameProcessor]] instead?

Reply:
fixed the comment